src/main/java/seedu/address/ui/InternshipApplicationCard.java

```diff
@@ -42,20 +50,49 @@
     @FXML
     private Label priority;
     @FXML
-    private Label status;
+    private FlowPane status;
 
     public InternshipApplicationCard(InternshipApplication internshipApplication, int displayedIndex) {
         super(FXML);
         this.internshipApplication = internshipApplication;
         id.setText(displayedIndex + ". ");
         company.setText(internshipApplication.getCompany().fullCompany);
-        phone.setText(internshipApplication.getPhone().value);
-        address.setText(internshipApplication.getAddress().value);
-        email.setText(internshipApplication.getEmail().value);
         role.setText(internshipApplication.getRole().fullRole);
-        priority.setText(Integer.toString(internshipApplication.getPriority().fullPriority));
-        applicationDate.setText(internshipApplication.getApplicationDate().toString());
-        status.setText(internshipApplication.getStatus().toString() + internshipApplication.getLastStageMessage());
+        Label statusLabel = getStatusLabel();
+        status.getChildren().add(statusLabel);
+    }
+
+    /**
+     * Gets the status label colored based on the status of the internship application.
+     *
+     * @return the status label colored based on the status of the internship application
+     */
+    private Label getStatusLabel() {
+        Status internshipApplicationStatus = internshipApplication.getStatus();
+        Label statusLabel = new Label(internshipApplicationStatus.toString().toLowerCase());
+        switch (internshipApplicationStatus) {
+        case WISHLIST:
+            statusLabel.setStyle("-fx-background-color: #cd70ff");
+            break;
+        case APPLIED:
+            statusLabel.setStyle("-fx-background-color: #209cee");
+            break;
+        case INTERVIEW:
+            statusLabel.setStyle("-fx-background-color: #22c65b");
+            break;
+        case OFFERED:
+            statusLabel.setStyle("-fx-text-fill: black; -fx-background-color: #ffdd57");
+            break;
+        case REJECTED:
+            statusLabel.setStyle("-fx-background-color: #ff3860; ");
+            break;
+        case GHOSTED:
+            statusLabel.setStyle("-fx-background-color: SLATEGRAY");
+            break;
```

Comment:
I think we should abstract out the magic value

Reply:
Ohh yea, good point